backend/src/packages/chaiNNer_standard/image_filter/correction/color_transfer.py

```diff
@@ -79,17 +79,45 @@ def color_transfer_node(
     if img_c == 4:
         alpha = img[:, :, 3]
     bgr_img = img[:, :, :3]
+
+    _, _, ref_img_c = get_h_w_c(ref_img)
+
+    ref_alpha = None
+    if ref_img_c == 4:
+        ref_alpha = ref_img[:, :, 3]
     bgr_ref_img = ref_img[:, :, :3]
 
+    # Don't process RGB data if the pixel is fully transparent, since
+    # such RGB data is indeterminate.
+    valid_rgb_indices = np.ones(img.shape[:-1], dtype=bool)
+    if alpha is not None:
+        valid_rgb_mask = alpha > 0
+        valid_rgb_indices = np.nonzero(valid_rgb_mask)
+
+    ref_valid_rgb_indices = np.ones(ref_img.shape[:-1], dtype=bool)
+    if ref_alpha is not None:
+        ref_valid_rgb_mask = ref_alpha > 0
+        ref_valid_rgb_indices = np.nonzero(ref_valid_rgb_mask)
```

Comment:
Why the `np.nonzero`? It works the same without it.

This would also allow us to change the type hints from `np.ndarray | tuple[np.ndarray, ...]` to just `np.ndarray`.

Reply:
Hmm, good question. I don't think I had a good reason for doing it this way. I'll try removing the `np.nonzero` and see if it still works OK on my end.

Reply:
Yes, seems to still work; thanks for the spot.